soc/arm/nordic_nrf/nrf54l/soc.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file SoC configuration macros for the Nordic Semiconductor NRF54L family processors.
+ */
+
+#ifndef _NORDICSEMI_NRF54L_SOC_H_
+#define _NORDICSEMI_NRF54L_SOC_H_
+
+#define __ICACHE_PRESENT    1
```

Comment:
why is this needed in soc.h?

Reply:
It's a CMSIS define that zephyr recently made necessary (if you enable ICACHE): https://github.com/zephyrproject-rtos/zephyr/pull/64978/files
For now it's missing in MDK and we cannot use `#include <cmsis_core_m_defaults.h>` as it causes compilation errors with nrfx.
Until new MDK introduces this define, it needs to be temporarily defined here.